interview-questions.md

```diff
@@ -5,57 +5,95 @@ Answer the following questions. First, without external resources. Challenge you
 
+  1. Start out with a class defintion for our component. It should extend from the React.Component class. 
+  2. Within this class is a single method, the render method. 
+  3. The render method will return a single HTML element. 
+  4. Our text will come after the return. 
+  5. Include final lines to define a container and tell React to render our component in the container.
 
 
 4. What is JSX?
 
   Your answer:
+  A language thats half Javascript, half HTML. You can use JSX within a React app. Allows you to use HTML tags within a React app in a Javascript file.
 
   Researched answer:
-
-
+  Syntax extension to JavaScript. JSX is a flavored HTML that behaves alsmost like HTML with a couple of exceptions, such as the className syntax rather than class. because class is reserved for HTML.
+  
+  JSX allows us to write HTML elements in JavaScript and place them in the DOM without any createElement() and/or appendChild() methods. JSX converts HTML tags into react elements. You are not required to use JSX, but JSX makes it easier to write React applications.
 
 5. What is state in React?
 
   Your answer:
+  Components can either hold states or not hold states. States are objects that store components dynamic data and determine how that component will behave.
 
   Researched answer:
-
+  React components has a built-in state object. The state object is where you store property values that belongs to the component. When the state object changes, the component re-renders. State is managed within the component (similar to variables declared within a function). 
 
 
 6. STRETCH: What is hoisting in JavaScript?
 
-  Your answer:
+  Your answer: (I had to Google this one)
+  Javascripts default behavior of moving variables/functions at the top of the currents cope. Declaring any variable or function in code is equal to declaring it at the top. 
+
+  Does this mean that the first thing JS does is run the defined variables and functions? Even if they might be at different scopes?
+
 
   Researched answer:
+  Hoisting is a JavaScript mechanism where variables and function declarations are moved to the top of their scope before code execution. Inevitably, this means that no matter where functions and variables are declared, they are moved to the top of their scope regardless of whether their scope is global or local.
 
 
 
 ## Looking Ahead: Terms for Next Week
 
 Research and define the following terms to the best of your ability.
 
-- React lifecycle methods
+- React lifecycle methods:
+Each component in React has a lifecycle which you can monitor and manipulate during its three main phases.
+
+The three phases are: Mounting, Updating, and Unmounting.
+
+Mounting: Mounting means putting elements into the DOM.
+Updating: A component is updated whenever there is a change in the component's state or props.
+Unmounting: The next phase in the lifecycle is when a component is removed from the DOM, or unmounting as React likes to call it.
```

Comment:
Totally cool. Theres A LOT more than just these, as you saw in your researched answer as well.